extensions/alpaca-trading/src/components/closePositionForm.tsx

```diff
@@ -0,0 +1,87 @@
+import { closePosition, Position } from '@api/positions';
+import { Action, ActionPanel, Alert, Color, confirmAlert, Form, Icon } from '@raycast/api';
+import { FormValidation, useForm } from '@raycast/utils';
+
+export interface ClosePositionFormValues {
+  quantity: string;
+  percentage: string;
+  kind: string;
+}
+
+export interface UseClosePositionFormProps {
+  position: Position;
+  revalidate: () => void;
+  pop: () => void;
+}
+
+export function ClosePositionForm({ position, revalidate, pop }: UseClosePositionFormProps) {
+  const { handleSubmit, itemProps, values } = useForm<ClosePositionFormValues>({
+    onSubmit: val =>
+      confirmAlert({
+        title: 'Are you sure?',
+        message: {
+          all: `Closing entire ${position.qty_available} quantities for ${position.symbol} position`,
+          qty: `Closing ${val.quantity} quantities for ${position.symbol} position`,
+          pc: `Closing ${val.percentage} % for ${position.symbol} position`,
+        }[val.kind],
+        icon: { source: Icon.TackDisabled, tintColor: Color.Red },
+        primaryAction: {
+          title: 'Yes',
+          style: Alert.ActionStyle.Destructive,
+          onAction: () =>
+            closePosition(position, { qty: `?qty=${val.quantity}`, pc: `?percentage=${val.percentage}` }[val.kind]).finally(() => {
+              pop();
+              setTimeout(revalidate, 1000);
+            }),
+        },
+        dismissAction: { title: 'No', style: Alert.ActionStyle.Cancel },
+      }),
+    initialValues: { kind: 'all' },
+    validation: {
+      kind: FormValidation.Required,
+      quantity: value => {
+        if (values.kind === 'qty' && Number.isNaN(Number(value)) && value!.split('.')[1]?.length > 9) {
```

Comment:
Should the value be strictly positive?

Reply:
Good catch, will resolve in a follow up commit.